lib/jekyll/assets/env.rb

```diff
@@ -111,6 +111,12 @@ def initialize(path, jekyll = nil)
         path ? super(path) : super()
         @jekyll = jekyll
 
+        # Analagous to Jekyll::Site @source and @destination paths
+        @cache_path = asset_config.fetch("cache", ".asset-cache")
+        if File.exist?(cache_path_in_source_dir = jekyll.in_source_dir(@cache_path))
+          @cache_path = cache_path_in_source_dir
+        end
+
```

Comment:
For back compat, resolve the cache dir against the source path first. Resolve the cache relative to the root dir if an old source-relative cache path doesn't already exist.